proof/invariant-abstract/RISCV64/ArchKernelInit_AI.thy

```diff
@@ -237,81 +275,197 @@ lemma kernel_mapping_slots_0x100[simp]:
                 pptrBase_def canonical_bit_def)
 
 lemma translate_address_kernel_window:
-  "\<lbrakk> pptr_base \<le> vref; vref < pptr_base + (1 << 30) \<rbrakk>\<Longrightarrow>
+  "\<lbrakk> pptr_base \<le> vref; vref < pptr_base + (1 << kernel_window_bits) \<rbrakk>\<Longrightarrow>
    translate_address riscv_global_pt_ptr vref (ptes_of init_A_st) = Some (addrFromPPtr vref)"
   apply (clarsimp simp: translate_address_def in_omonad pt_lookup_target_def
                         pt_lookup_slot_from_level_def)
   apply (simp add: ptes_of_init_A_st_global[THEN fun_cong] init_global_pt_def global_pte_def pte_ref_def)
-  apply (simp add: kernel_window_1G is_aligned_pt_slot_offset_pte)
+  apply (simp add: kernel_window_bits_table_index is_aligned_pt_slot_offset_pte)
   apply (simp add: bit_simps addr_from_ppn_def shiftl_shiftl)
   apply (simp add: ptrFromPAddr_def addrFromPPtr_def)
   apply (simp add: pptrBaseOffset_def paddrBase_def)
-  apply (simp add: pt_bits_left_def bit_simps level_defs)
+  apply (simp add: pt_bits_left_def bit_simps level_defs elf_index_value toplevel_bits_def)
   apply (rule conjI)
    apply (rule is_aligned_add)
     apply (simp add: mask_def)
    apply (simp add: pptrBase_def canonical_bit_def is_aligned_def)
-  apply (simp add: pptr_base_def)
+  apply (simp add: pptr_base_def kernel_window_bits_def)
   apply (simp add: pptrBase_def neg_mask_le_high_bits flip: NOT_mask)
   apply (subst word_plus_and_or_coroll; simp add: canonical_bit_def word_size mask_def)
   apply word_bitwise
   apply clarsimp
   done
 
-lemma elf_window_1M:
-  "\<lbrakk> kernel_elf_base \<le> vref; vref < kernel_elf_base + (1 << 20) \<rbrakk> \<Longrightarrow>
-    table_index (pt_slot_offset max_pt_level riscv_global_pt_ptr vref) = 0x1FE"
-  apply (simp add: table_index_riscv_global_pt_ptr)
-  apply (simp add: bit_simps kernel_elf_base_def kernelELFBase_def kernelELFPAddrBase_def
-                   Kernel_Config.physBase_def)
+lemma is_page_aligned_physBase: (* FIXME MK: move to Kernel_Config_Lemmas *)
+  "is_aligned physBase pageBits"
+  by (simp add: Kernel_Config.physBase_def pageBits_def is_aligned_def)
+
+lemma is_page_aligned_kernelELFPAddrBase: (* FIXME MK: does not mention physBase, but could be moved up *)
+  "is_aligned kernelELFPAddrBase pageBits"
+  unfolding kernelELFPAddrBase_def
+  by (fastforce intro!: is_aligned_add is_page_aligned_physBase
+                simp: kernelELFPAddrBase_def pageBits_def is_aligned_def)
+
+(* FIXME MK: move *)
+lemma aligned_mask_le_mask_minus:
+  "\<lbrakk> is_aligned x m; m \<le> n; n < LENGTH('a)\<rbrakk> \<Longrightarrow> x && mask n \<le> mask n - mask m" for x :: "'a::len word"
+  by (metis and_mask_less' is_aligned_after_mask is_aligned_neg_mask_eq'
+           mask_2pm1 mask_sub neg_mask_mono_le word_less_sub_le)
+
+(* FIXME MK: rename, move *)
+lemma aligned_mask_plus_bounded:
```

Comment:
have done the rename already, but still should be moved to the Word_LIb staging area